reactor-netty-core/src/main/java/reactor/netty/resources/DefaultPooledConnectionProvider.java

```diff
@@ -454,7 +452,7 @@ public void onStateChange(Connection connection, State newState) {
 				                                 pool.metrics().acquiredSize(),
 				                                 pool.metrics().idleSize());
 				                     }
-				                     onTerminate.onComplete();
+				                     onTerminate.emitEmpty(Sinks.EmitFailureHandler.FAIL_FAST);
```

Comment:
I think this can also be `Sinks.unsafe().empty()` with `tryEmit()` since the signals are serialized already. See [this mention](https://github.com/reactor/reactor-core/blob/332b18ba2cf25ff9d93d01456d6ec79f30a61ccc/reactor-core/src/main/java/reactor/core/publisher/Sinks.java#L99-L101) for example.

Reply:
please check